### ISSUE TITLE
Add support for Typescript based React components

### ISSUE THREAD

**Issue by jimidle** [idyll-lang/idyll#788]
**Is your feature request related to a problem? Please describe.**
Many graphical components are now written using typescript .tsx, and it can be awkward to create components that interface to them from .js - though I admit that a better React programmer than I am may find that to be fairly trivial. 

**Describe the solution you'd like**
Add support for components written in Typescript, with a few examples like the d3 examples and one practical interface such as one of the React graphing libraries.

**Describe alternatives you've considered**
Learning how to interact with external tsx based components using .js.

**Additional context**
Everything seems to be written using Typescript now and Idyll is a little behind. It would definitely be an upgrade to support .tsx based components.

**Comment by mathisonian** [comment]
Agreed this would be a nice thing to support. Under the hood Idyll uses esbuild, which does have [support for loading typescript files](https://esbuild.github.io/content-types/#typescript). I believe we would need to update our component resolution logic to accept `.ts` or `.tsx` files (see [here](https://github.com/idyll-lang/idyll/blob/master/packages/idyll-cli/src/resolvers/components.js#L99-L101)). 

I would start by updating the component resolver, and testing if additional configuration changes to the [esbuild invocation](https://github.com/idyll-lang/idyll/blob/master/packages/idyll-cli/src/pipeline/bundle-js.js#L35-L55) are needed.

**Comment by jimidle** [comment]
I have this working now, after some wrangling to work out what the flow through `babel` to `esbuild` is. However, in `parse.js` there is an `exports.getHTML = ...` that `trys` a `require` on each of the files in the `.idyll/components/0` and trys an `await import` if that fails. The file that is in the temporary components directory is the `.tsx` file, which cannot be imported of course. 

Now, I am not quite sure what this code is doing for server side rendering. Perhaps you could briefly enlighten me?

So my questions are:

1. Should I try to get babel to transform to .`js`, keep that file, and put that in the temporary components directory instead of the `.tsx` file?
2. Assume that whatever this code is doing, that components coming from `.ts` and `.tsx files don't need to be processed here, and just skip them? 

With this resolved, I can create a PR for you to take a look at.

**Comment by jimidle** [comment]
I have completed this work, but I cannot create a new pull request because the pull request for #748 requires merging first. Perhaps moving to use git flow as system would help with this. I guess that I should have done this myself.

I don't want to cancel the existing pull request and create a new one because then the PR will cover two features, which is not correct. 

I will await your approval of the previous PR then I can submit the PR for this one. Github will notify me when you have done this I suspect.

**Comment by jimidle** [comment]
No worries - when I take advantage of someone else's good work, I think it is only polite to give a little bit back.
